src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -138,6 +138,13 @@ protected void AddVariablesToEnvironment(bool excludeNames = false, bool exclude
                 string formattedKey = (pair.Key ?? string.Empty).Replace('.', '_').ToUpperInvariant();
                 AddEnvironmentVariable(formattedKey, pair.Value);
 
+                // for maintaining back compat variable names replace space with underscore
+                string backCompatKey = formattedKey.Replace(' ', '_');
```

Comment:
by calling this backCompat, it's implying that it's not something we want but we're keeping in the code solely for compat.  but I believe this was just an omission.